CommonTools/MVAUtils/src/GBRForestTools.cc

```diff
@@ -117,6 +120,15 @@ namespace {
   }
 
   std::unique_ptr<GBRForest> init(const std::string& weightsFileFullPath, std::vector<std::string>& varNames) {
+    //
+    // Load weights file, for ROOT file
+    //
+    if (reco::details::hasEnding(weightsFileFullPath, ".root")) {
+      TFile gbrForestFile(weightsFileFullPath.c_str());
+      // the pointer is free to be taken over
+      return std::unique_ptr<GBRForest>(reinterpret_cast<GBRForest*>(gbrForestFile.Get("gbrForest")));
```

Comment:
as proposed in mattermost a semantically safer version would be
```suggestion
      std::unique_ptr<GBRForest> up(gbrForestFile.Get<GBRForest>("gbrForest"));
      gbrForestFile.Close("nodelete");
      return up;
```

I can confirm that it works at least from MEM_TOTAL pov
https://slava77sk.web.cern.ch/slava77sk/reco/cgi-bin/igprof-navigator/CMSSW_11_2_0_pre9-sign1110-57c942e.136.88811.step2.IgProf.1.MEM_LIVE/174